### PR TITLE
better unique id approach for RevolveGenerator

### LINKED ISSUE #28401: Unique ID Issue in RevolveGenerator
### Bug Description

When we set a `unique_id()` before adding an object to a mesh, the mesh uses that id to update it's cache of the max `unique_id()` (on that processor, if it's distributed), so that `parallel_max_unique_id()` can be O(1) instead of O(N).

When we set a `unique_id()` *after* adding an object to a mesh, the mesh has no idea anything happened.  `parallel_max_unique_id()` is wrong.  Any code that uses it (including XYZDelaunay, and ironically RevolveGenerator) afterward is wrong.

There's a comment saying that RevolveGenerator was "Inspired by AdvancedExtruderGenerator", which used to have the same bug.  The simplest fix is just a `set_next_unique_id` call that takes into account the generator's formula for setting new unique ids.

_Originally posted by @roystgnr in https://github.com/idaholab/moose/issues/28298#issuecomment-2292446675_
          

### Steps to Reproduce

See Bug Description

### Impact

Followup MGs may get wrong starting unique id and then the new unique id assignment would be messed.

### [Optional] Diagnostics

_No response_

### PR DESCRIPTION
**Pull request by miaoyinb**
closes #28401

### CODE REVIEW

**Comment by miaoyinb** [comment]
@roystgnr FYI

File: framework/src/meshgenerators/AdvancedExtruderGenerator.C [hunk 1]
```diff
@@ -1024,12 +1024,14 @@ AdvancedExtruderGenerator::generate()
     }
   }
 
+#ifdef LIBMESH_ENABLE_UNIQUE_ID
```
Comment (roystgnr) [inline framework/src/meshgenerators/AdvancedExtruderGenerator.C:1027]: I approve as a matter of principle, but I don't think MOOSE has compiled without unique_id support for years.

**Review by roystgnr — DISMISSED** [review]

**Comment by miaoyinb** [comment]
Okay, I will double check. 

Do you happen to have an input file that leads to duplicate ids?

**Comment by roystgnr** [comment]
There's one in the `Modules debug` recipe failure:

```
reactor/test:meshgenerators/revolve_generator.test/simple_quad8_2: Working Directory: /data/civet0/build/moose/modules/reactor/test/tests/meshgenerators/revolve_generator
reactor/test:meshgenerators/revolve_generator.test/simple_quad8_2: Running command: /data/civet0/build/moose/modules/combined/combined-dbg -i revolve_2d.i --mesh-only revolve_2d_quad8_2.e Mesh/gmg/elem_type=QUAD8 Mesh/rg/axis_direction="-1.0 1.0 0.0" --error --error-override --timing Outputs/perf_graph=true --libtorch-device cpu
reactor/test:meshgenerators/revolve_generator.test/simple_quad8_2: 
reactor/test:meshgenerators/revolve_generator.test/simple_quad8_2: 
reactor/test:meshgenerators/revolve_generator.test/simple_quad8_2: *** ERROR ***
reactor/test:meshgenerators/revolve_generator.test/simple_quad8_2: 
reactor/test:meshgenerators/revolve_generator.test/simple_quad8_2: Stack frames: 15
reactor/test:meshgenerators/revolve_generator.test/simple_quad8_2: 0: libMesh::print_trace(std::ostream&)
reactor/test:meshgenerators/revolve_generator.test/simple_quad8_2: 1: libMesh::MacroFunctions::report_error(char const*, int, char const*, char const*, std::ostream&)
reactor/test:meshgenerators/revolve_generator.test/simple_quad8_2: 2: libMesh::MeshTools::libmesh_assert_valid_unique_ids(libMesh::MeshBase const&)
reactor/test:meshgenerators/revolve_generator.test/simple_quad8_2: 3: libMesh::MeshBase::prepare_for_use()
```

The failing line is an assertion that an iterator over nodes doesn't see a unique_id that it's already seen on a previous node or element.

**Review by roystgnr — APPROVED** [review]

**Pull request merged**